apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts

```diff
@@ -83,6 +86,26 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
   }
 
   public ngOnInit() {
+    this.holdings$ = this.dataService
+      .fetchPortfolioDetails({
+        filters: [
+          {
+            type: 'ACCOUNT',
+            id: this.data.accountId
+          }
+        ]
+      })
+      .pipe(
+        map((d) =>
+          Object.values(d.holdings).map(function (x) {
```

Comment:
Can you move that transformation directly to `fetchPortfolioDetails()`?

You can then even remove the same code in `HoldingsPageComponent` 🙂